base/array.jl

```diff
@@ -1439,10 +1439,15 @@ you expect that you're going to have to push a lot of values onto `s`, you can a
 the cost of incremental reallocation by doing it once up front; this can improve
 performance.
 
-If `first` is true, then the reserved space is from the start of the collection, for ordered
-collections. Supplying this keyword may result in an error if the collection is nor ordered
+If the collection is ordered and `first` is `true`, then any additional space is reserved
```

Comment:
Here, I slightly prefer the original formulation, but the second part with `pushfirst!` seems very helpful. Also the condition for the collection to be ordered is kind of loose. For example `first=true` probably have no effect for `OrderedDict`.